### PR TITLE
Implement connect_forever in util. Related to #73

### PR DESCRIPTION
**Pull request by Alexei-Kornienko**
This enables us to connect socket to server that doesn't exist yet.

### CODE REVIEW

**Comment by Alexei-Kornienko** [comment]
@TheButlah pls review when you have time. This MR partially fixes #73.

File: src/transport/ipc.rs [hunk 1]
```diff
@@ -15,8 +15,8 @@ use crate::ZmqResult;
 use futures::{select, FutureExt};
 use std::path::{Path, PathBuf};
 
-pub(crate) async fn connect(path: PathBuf) -> ZmqResult<(FramedIo, Endpoint)> {
-    let raw_socket = UnixStream::connect(&path).await?;
+pub(crate) async fn connect(path: &PathBuf) -> ZmqResult<(FramedIo, Endpoint)> {
```
Comment (TheButlah) [inline src/transport/ipc.rs:18]: If we don't need ownership of the `PathBuf`, then this should just be an `&Path`

**Review by TheButlah — CHANGES_REQUESTED** [review]
In order to close #73 , we need connect to return immediately, queuing messages until we run out of memory (or hit the currently unimplemented high water mark). Right now the code instead blocks until the connection is successful. This is an improvement, and we can definitely merge it in this state, but let's not close #73 until we implement HWM functionality and make the connect function non-blocking

File: src/transport/ipc.rs (hunk 1 above)
Comment (Alexei-Kornienko) [reply src/transport/ipc.rs:18]: Thanks for the hint. I'll fix it

**Comment by Alexei-Kornienko** [comment]
I agree. I marked this MR as related cause some additional work is needed to make it fully spec compliant.

File: src/transport/ipc.rs (hunk 1 above)
Comment (Alexei-Kornienko) [reply src/transport/ipc.rs]: Done

**Review by TheButlah — APPROVED** [review]

**Pull request merged**